qiskit/transpiler/passes/scheduling/calibration_creators.py

```diff
@@ -0,0 +1,222 @@
+
+    def supported(self, node_op: DAGNode) -> bool:
+        """
+        Args:
+            node_op: The node from the dag dep.
+
+        Returns:
+            match: True if the node is a RZXGate.
+        """
+        return isinstance(node_op, RZXGate)
+
+    @staticmethod
+    def rescale_cr_inst(instruction: Play, theta: float, sample_mult: int = 16) -> Play:
+        """
+        Args:
+            instruction: The instruction from which to create a new shortened or lengthened pulse.
+            theta: desired angle, pi/2 is assumed to be the angle that the pulse in the given
+                play instruction implements.
+            sample_mult: All pulses must be a multiple of sample_mult.
+
+        Returns:
+            Play: The play instruction with the stretched compressed GaussianSquare pulse.
+
+        Raises:
+            QiskitError: if the pulses are not GaussianSquare.
+        """
+        pulse_ = instruction.pulse
+        if isinstance(pulse_, GaussianSquare):
+            amp = pulse_.amp
+            width = pulse_.width
+            sigma = pulse_.sigma
+            n_sigmas = (pulse_.duration - width) / sigma
+
+            # The error function is used because the Gaussian may have chopped tails.
+            gaussian_area = abs(amp) * sigma * np.sqrt(2 * np.pi) * math.erf(n_sigmas)
+            area = gaussian_area + abs(amp) * width
+
+            target_area = abs(theta) / (np.pi / 2.) * area
+            sign = theta / abs(theta)
+
+            if target_area > gaussian_area:
+                width = (target_area - gaussian_area) / abs(amp)
+                duration = math.ceil((width + n_sigmas * sigma) / sample_mult) * sample_mult
+                return Play(GaussianSquare(amp=sign*amp, width=width, sigma=sigma,
+                                           duration=duration), channel=instruction.channel)
+            else:
+                amp_scale = sign * target_area / gaussian_area
+                duration = math.ceil(n_sigmas * sigma / sample_mult) * sample_mult
+                return Play(
+                    GaussianSquare(amp=amp * amp_scale, width=0, sigma=sigma, duration=duration),
+                    channel=instruction.channel)
+        else:
+            raise QiskitError('RZXCalibrationBuilder only stretches/compresses GaussianSquare.')
+
+    def get_calibration(self, params: List, qubits: List) -> Schedule:
+        """
+        Args:
+            params: Parameters of the RZXGate(theta). I.e. params[0] is theta.
+            qubits: List of qubits for which to get the schedules. The first qubit is
+                the control and the second is the target.
+
+        Returns:
+            schedule: The calibration schedule for the RZXGate(theta).
+
+        Raises:
+            QiskitError: if the the control and target qubits cannot be identified.
+        """
+        theta = params[0]
+        q1, q2 = qubits[0], qubits[1]
+        cx_sched = self._inst_map.get('cx', qubits=(q1, q2))
```

Comment:
This is unconditionally accessing the calibration of 'cx', which may not exist in general (actually I think IBM devices are transitioning to reporting `ecr` (echo cross resonance) as native gate which doesn't have the pre-rotations of cx).
So instead of letting this raise an error, I think it's good to do a check that `cx` exists and if not raise an error saying this pass currently works from an assumption of cx being present.

Reply:
Good point I added a check to see if we have `cx`. Supporting `ecr` will be a nice extension.